### PR TITLE
HOCS-5278: add field restriction list

### PR DESCRIPTION
**Pull request by LiamMacP**
This change causes fields to be removed from the case data on restricted
cases by exemption rather than an inclusion policy. This is done
through the `RestrictedField[Service/Repository]` instead of  behaviour
added previously to the UserPermissionService that communicated with
`hocs-info-service`.

As we are handling changes to collections whereby the objects within a
mutable - this change removes the need for helper static classes.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/RestrictedFieldPermissions.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.Map;
+import java.util.Set;
+
+public class RestrictedFieldPermissions {
+
+    @JsonValue
+    public Map<String, FieldAccessLevels> caseFieldAccessLevels;
+
+    @JsonCreator
```
Comment (nickmcmahon01) [inline src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/RestrictedFieldPermissions.java:15]: Thankyou for un-lombocking

File: src/main/resources/config/restricted-fields.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+  "SMC" : {
```
Comment (nickmcmahon01) [inline src/main/resources/config/restricted-fields.json:2]: I think this 'soft link' is fine looking through the document removes a load of UUIDs too

File: src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/RestrictedFieldPermissions.java (hunk 1 above)
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/RestrictedFieldPermissions.java:15]: I had it with Lombok but JsonCreator actually needs an AllArgsConstructor to work

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/GetCaseSummaryResponse.java [hunk 3]
```diff
@@ -93,28 +92,6 @@ public static GetCaseSummaryResponse from(CaseSummary caseSummary) {
                 caseSummary.getActions(),
                 caseSummary.getSuspended()
         );
-
-
-    }
-
-    protected void replaceAdditionalFields(List<AdditionalFieldDto> additionalFieldDtoList) {
```
[the remark this answers is not in the document]
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/GetCaseSummaryResponse.java:100]: They were helpers to just clear out collections, a bit pointless when while the properties were immutable, the items in could just be wiped.

File: src/main/resources/config/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+```json
+{
+ <<CASE_TYPE>>: {
+  <<PERMISSION_LEVEL>>: [
+   <<FIELD_NAMES>>
+  ] 
+ }
+}
+```
```
Comment (nickmcmahon01) [inline src/main/resources/config/README.md:22]: I'm new to the project, how to do I add one for a new case type?

**Review by nickmcmahon01 — APPROVED** [review]

File: src/main/resources/config/README.md (hunk 4 above)
Comment (LiamMacP) [reply src/main/resources/config/README.md:22]: https://cheatography.com/gaston/cheat-sheets/json/

**Pull request merged**